sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteTables.java

```diff
@@ -284,19 +305,25 @@ private void load(
           }
           return;
         case UNKNOWN:
-          LOG.info("Load job {} finished in unknown state: {}", jobRef, loadJob.getStatus());
-          throw new RuntimeException(
-              String.format(
-                  "UNKNOWN status of load job [%s]: %s.",
-                  jobId, BigQueryHelpers.jobToPrettyString(loadJob)));
-        case FAILED:
+          // This might happen if BigQuery's job listing is slow. Retry with the same
+          // job id.
           LOG.info(
-              "Load job {} failed, {}: {}",
+              "Load job {} finished in unknown state: {}: {}",
               jobRef,
-              (i < BatchLoads.MAX_RETRY_JOBS - 1) ? "will retry" : "will not retry",
-              loadJob.getStatus());
+              loadJob.getStatus(),
+              (i < maxRetryJobs - 1) ? "will retry" : "will not retry");
           lastFailedLoadJob = loadJob;
           continue;
+        case FAILED:
+          lastFailedLoadJob = loadJob;
+          jobId = BigQueryHelpers.getRetryJobId(jobId, projectId, bqLocation, jobService).jobId;
+          LOG.info(
```

Comment:
Should we retry actually failed jobs ? This might have been due to a legitimate client error, no ? Should we refer to error code/message before retrying ?

Reply:
I guess it might be, but what other option do we have? Not retrying the job is potentially data loss.